ibis/impala/compiler.py

```diff
@@ -217,6 +317,13 @@ def _window(translator, expr):
             len(window._order_by) == 0):
         window = window.order_by(window_op.args[0])
 
+    # Time ranges need to be converted to microseconds.
+    if window.how == 'range':
+        order_by_types = [type(x.op().args[0]) for x in window._order_by]
+        time_range_types = (ir.TimeColumn, ir.DateColumn, ir.TimestampColumn)
+        if any([col_type in time_range_types for col_type in order_by_types]):
```

Comment:
No need for the `[]` here in the `any` call, you can write this as `any(x for x in y)`.